app/src/main/java/com/pmatuki/wowpapers/view/detail/DetailActivity.kt

```diff
@@ -58,10 +60,23 @@ class DetailActivity : AppCompatActivity() {
                     binding.apply {
                         layoutViewLoading.visibility = View.GONE
                         cardView.visibility = View.VISIBLE
+                        applyButton.isEnabled = true
                     }
                     wallpaperDrawable = state.imageDrawable
                     this@DetailActivity.binding.backgroundImage.setImageDrawable(wallpaperDrawable)
                 }
+                DetailViewState.Applying -> {
```

Comment:
You mentioned that you want me to review the states from this Activity and does look good to me. I prefer to have a more complex state handling, with multiple options like you did here, instead of having more than one state. In my personal experience, always prefer to have a Single Source of Truth, like you did here.

One way of maybe make the code simpler for the UI (specially if you will have more states here) is creating a `data class` containing all the view properties from your screen. You would build that component there on the ViewModel and only pass the class to be rendered here.

For example:
```
data class DetailViewState(
    val loadingVisibility: Int,
    val cardVisibility: Int,
    val applyButtonEnabled: Boolean
)
```
Then your ViewModel will build this object based on the states and the Activity will only do something like:
```
fun setupState(state: DetailViewState) {
    layoutViewLoading.visibility = state.loadingVisibility
    cardView.visibility = state.cardVisibility
    applyButton.isEnabled = state.applyButtonEnabled
}
```

However, at the moment I prefer the implementation you did here. I don't like to give too much knowledge about the view to the ViewModel. But in very complex screens, this is an alternative.